directory/datadog_checks/directory/directory.py

```diff
@@ -46,12 +50,21 @@ def check(self, _):
                 "Either directory '{}' doesn't exist or the Agent doesn't "
                 "have permissions to access it, skipping.".format(self.config.abs_directory)
             )
-
-            if not self.config.ignore_missing:
+            # report missing directory and skip check
+            if self.config.report_missing:
+                self.log.info('service check SERVICE_DIRECTORY_EXISTS')
+                self.service_check(name=SERVICE_DIRECTORY_EXISTS, status=self.WARNING, message=msg)
+            # or ignore it
+            elif self.config.ignore_missing:
+                pass
+            else:
                 raise ConfigurationError(msg)
```

Comment:
I'm not sure this should be a ConfigurationError or a general check exception, a directory disapearing is not necessary a configuration error plus those should only be raise at `__init__`.

Reply:
moving to `CheckException`